src/esbmc/bmc.cpp

```diff
@@ -770,95 +786,82 @@ smt_convt::resultt bmct::multi_property_check(
-            // - locaiton: ideally, the loc is enough for distinguishuing claims.
-            // - comment: we add a claim number prefix in comment. Note that the unwinding assertions which are not processed by goto-cov.
-            // Another reason is that we need to display this info in goto-coverage-claims
-
-            cmt_loc = step.comment + "\t" + loc;
-          }
-
-        bool is_unverified = false;
-
-        if(is_goto_cov)
-        {
-          reached_mul_claims.emplace(cmt_loc);
-          is_unverified = true;
-        }
-        else
+      goto_tracet goto_trace;
+      build_goto_trace(local_eq, runtime_solver, goto_trace, is_compact_trace);
+
+      // Store the comment and location of the assertion
+      // to avoid double verifying the claims that are already verified
+      std::string cmt_loc = "";
+
+      for(const auto &step : goto_trace.steps)
+        if(step.type == goto_trace_stept::ASSERT)
         {
-          // the ins is true if the element was actually inserted
-          auto [it, ins] = reached_claims.emplace(cmt_loc);
-          is_unverified = ins;
+          // since we only handle one claim at a time
+          // we will/should not overwrite the loc
+          assert(cmt_loc == "");
+          std::string loc;
+          if(step.pc->location.is_nil())
+            loc = "nil";
+          else
+            loc = step.pc->location.as_string();
+
+          // we use the "comment + location" to distinguish each claim
+          // e.g. "Claim x: ... location line y"
+          // x is unique. However, the unwinding asserts do not have these Claim x prefixes.
+          // Therefore we add the location behind, as the line number y for each unwinding assert is different.
+          cmt_loc = step.comment + "\t" + loc;
         }
 
-        if(is_unverified || options.get_bool_option("keep-verified-claims"))
+      bool is_unverified = false;
+
+      if(is_goto_cov)
+      {
+        reached_mul_claims.emplace(cmt_loc);
+        is_unverified = true;
+      }
+      else
+      {
+        // the ins is true if the element was actually inserted
+        auto [it, ins] = reached_claims.emplace(cmt_loc);
+        is_unverified = ins;
+      }
+
+      if(is_unverified || options.get_bool_option("keep-verified-claims"))
+      {
+        std::string output_file = options.get_option("cex-output");
+        if(output_file != "")
         {
-          std::string output_file = options.get_option("cex-output");
-          if(output_file != "")
-          {
-            std::ofstream out(fmt::format("{}-{}", output_file, ce_counter++));
-            show_goto_trace(out, ns, goto_trace);
-          }
-          std::ostringstream oss;
-          log_fail("\n[Counterexample]\n");
-          show_goto_trace(oss, ns, goto_trace);
-          log_result("{}", oss.str());
-          final_result = result;
+          std::ofstream out(fmt::format("{}-{}", output_file, ce_counter++));
```

Comment:
Sorry, I'm seeing this just now. The help message for `--cex-output arg` says
> save the counterexample into a file

Appearently that's not what's happening here. Could you please update the help message to account for
1. the fact that with `--multi-property` multiple cex-output files are being created, and
2. that in this case the format is `arg-N` where N is a decimal increasing from zero(?).

I'm not often thinking about Windows, but isn't appending some string to filenames given by Windows users counter-intuitive? If this is just a debug option, this remark is not super important. But imagining a user accustomed to Windows passing `--multi-property --cex-output cex.txt` the result might be pretty confusing for this user, particularly so when displaying filename extensions has been turned off.

Reply:
I changed it to the prefix format. So now it will generate files like `0-cex.txt`, `1-cex.txt`.

I also updated the help message:
```
  --cex-output arg                      save the counterexample into a file or,
                                        in multi-property mode, multiple files
                                        with prefix 'N-' where 'N' is a decimal
                                        increasing from zero
```

Reply:
Thanks!